### PR TITLE
Hide login status from login required APIs

### LINKED ISSUE #274: Validators shouldn't expose hide/show validator
Instead of hiding and showing the validator outside the API, we should just call onLogged INSIDE the API

DUH!!

### PR DESCRIPTION
**Pull request by braxtonhall**
resolves #274 

came to me in a dream

this was literally the part of the code I hated the most. So happy this is fixed.

### CODE REVIEW

File: src/ts/content/extensions/author/authorPage/authorUI.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ const currentTagViewButtons = ({onPush, onSync, onEdit}: ButtonHandlers, getTags
 	);
 	const onEditButton = createTagButton("Edit", "img/edit.gif", onEdit);
 
-	onLogged({
+	void onLogged({
```
Comment (adirar111) [inline src/ts/content/extensions/author/authorPage/authorUI.ts:85]: i read the docs and i still don't understand what this does lol

File: src/ts/content/extensions/tags.ts [hunk 2]
```diff
@@ -40,9 +40,10 @@ onFormRender((form, forEachElement, onSave, offSave) => {
 			textAreaId: tagsTextAreaId,
 			transform: (tags: string[]) => tags.join(", "),
 			delimiter: ", ",
+			requiresLogged: true,
 		});
 
-		const {showFinder: showAncestors, hideFinder: hideAncestors} = appendFinder<string[]>({
+		insertFinder<string[]>({
```
Comment (adirar111) [inline src/ts/content/extensions/tags.ts:46]: oh cool so now we just call this instead of unpacking stuff and putting them back into `onLogged`?

**Review by adirar111 — APPROVED** [review]
you need to stop dreaming about programming

File: src/ts/content/extensions/author/authorPage/authorUI.ts (hunk 1 above)
Comment (braxtonhall) [reply src/ts/content/extensions/author/authorPage/authorUI.ts:85]: it casts the return value of the function call to `void`

Useful if you want to communicate "no i didn't forget to await this call -- this was on purpose that the promise is floating"

File: src/ts/content/extensions/tags.ts (hunk 2 above)
Comment (braxtonhall) [reply src/ts/content/extensions/tags.ts:46]: yesssss

**Pull request merged**